plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1278,4 +1282,50 @@ public void testDeleteWithVarcharPredicate()
     {
         throw new SkipException("This is implemented by testDeleteWithVarcharEqualityPredicate");
     }
+
+    @Test(dataProvider = "testInsertBatchSizeSessionProperty")
+    public void testInsertBatchSizeSessionProperty(Integer batchSize, Integer numberOfRows)
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new SkipException("CREATE TABLE is required for insert_batch_size test but is not supported");
+        }
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "insert_batch_size", batchSize.toString())
+                .build();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_insert_batch_size",
+                "(a varchar(36), b bigint)")) {
+            String values = String.join(",", makeValuesForInsertBatchSizeSessionPropertyTest(numberOfRows));
+            assertUpdate(session, "INSERT INTO " + table.getName() + " (a, b) VALUES " + values, numberOfRows);
+            assertQuery("SELECT COUNT(*) FROM " + table.getName(), format("VALUES %d", numberOfRows));
+        }
+    }
+
+    private static List<String> makeValuesForInsertBatchSizeSessionPropertyTest(int numberOfRows)
+    {
+        List<String> result = new ArrayList<>(numberOfRows);
+        for (int i = 0; i < numberOfRows; i++) {
+            result.add(format("('%s', %d)", UUID.randomUUID(), ThreadLocalRandom.current().nextLong()));
+        }
+        return result;
+    }
+
+    @DataProvider(name = "testInsertBatchSizeSessionProperty")
+    public static Object[][] batchSizeAndNumberOfRowsForInsertBatchSizePropertyTest()
```

Comment:
@sergey-melnychuk Why do we need name redefinition? I think if we remove 
```
@DataProvider(name = "testInsertBatchSizeSessionProperty")
```
line and rename 
`batchSizeAndNumberOfRowsForInsertBatchSizePropertyTest` -> 
`batchSizeAndTotalNumberOfRowsToInsertDataProvider` 
code would be more readable.

Reply:
Naming things, right? :) This PR is merged already, seems it was readable enough. I suggest you submit a PR if you're unhappy with current readability - I will gladly approve it, even though it is not mine approval that you'll need anyway :)

Reply:
https://github.com/trinodb/trino/pull/8498